core/chains/evm/txmgr/orm.go

```diff
@@ -42,6 +42,78 @@ type orm struct {
 
 var _ ORM = (*orm)(nil)
 
+// Structs saved in database should be internal to
+// the ORM and distinct from generic params
+// so that modifying generic params does neccessitate
+// modification of underlying column types
+// (or runtime errors by mistake)
+type dbReceipt struct {
```

Comment:
Another way of explaining: the generic parameters in `txmgrtypes.TxStorageService[common.Address, big.Int, common.Hash, NewTx, evmtypes.Receipt, EthTx, EthTxAttempt, int64, int64]` will change so we don't want that change to affect the underlying db column types

Ex: dbReceipt is equivalent (for now) txmgrtypes.Receipt[evmtypes.Receipt, common.Hash] but we don't want to use that type directly to save to database because we could easily change those types in the future